### PR TITLE
8317868: Add @sealedGraph to MethodHandleDesc and descendants

### PR DESCRIPTION
**Pull request by minborg**
This PR proposes to add  @sealedGraph to MethodHandleDesc and descendants.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8317868](https://bugs.openjdk.org/browse/JDK-8317868): Add @sealedGraph to MethodHandleDesc and descendants (**Sub-task** - P4)

### Reviewers
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - **Reviewer**)
 * [Hamlin Li](https://openjdk.org/census#mli) (@Hamlin-Li - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16137/head:pull/16137` \
`$ git checkout pull/16137`

Update a local copy of the PR: \
`$ git checkout pull/16137` \
`$ git pull https://git.openjdk.org/jdk.git pull/16137/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16137`

View PR using the GUI difftool: \
`$ git pr show -t 16137`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16137.diff">https://git.openjdk.org/jdk/pull/16137.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16137#issuecomment-1757053348)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/constant/DirectMethodHandleDesc.java [hunk 1]
```diff
@@ -46,6 +46,7 @@
  * {@link MethodHandle}.  A {@linkplain DirectMethodHandleDesc} corresponds to
  * a {@code Constant_MethodHandle_info} entry in the constant pool of a classfile.
  *
+ * @sealedGraph
```
Comment (liach) [inline src/java.base/share/classes/java/lang/constant/DirectMethodHandleDesc.java:49]: DMH desc has no public implementation subtypes. A `@sealedGraph` would probably just render a single root node, which is not informative.

Comment (minborg) [reply src/java.base/share/classes/java/lang/constant/DirectMethodHandleDesc.java:49]: Well spotted. Thanks.

**Comment by minborg** [comment]
Any reviewers on this very small PR?

**Review by JornVernee — APPROVED** [review]

**Review by Hamlin-Li — APPROVED** [review]

**Pull request closed without merging**